gateway-js/src/__tests__/integration/scope.test.ts

```diff
@@ -0,0 +1,116 @@
+import gql from 'graphql-tag';
+import { execute } from '../execution-utils';
+
+import {
+  astSerializer,
+  queryPlanSerializer,
+} from 'apollo-federation-integration-testsuite';
+
+expect.addSnapshotSerializer(astSerializer);
+expect.addSnapshotSerializer(queryPlanSerializer);
+
+describe('scope', () => {
+  it("doesn't wrap inline fragments with the supertype when @include is used", async () => {
+    const query = `#graphql
+    query GetProducts {
+      topProducts {
+        name
+        ... on Shoe @include(if: true) {
+          rating
+        }
+        ... on Car {
+          rating
+        }
+      }
+    }
+  `;
+
+    const { queryPlan, errors } = await execute({ query }, [
+      {
+        name: 'products',
+        typeDefs: gql`
+          extend type Query {
+            topProducts: [Product]
+          }
+
+          interface Product {
+            name: String
+          }
+
+          type Shoe implements Product @key(fields: "upc") {
+            upc: String
+            name: String
+          }
+
+          type Car implements Product @key(fields: "upc") {
+            upc: String
+            name: String
+          }
+        `,
+      },
+      {
+        name: 'reviews',
+        typeDefs: gql`
+          extend type Shoe @key(fields: "upc") {
+            upc: String @external
+            rating: Int
+          }
+
+          extend type Car @key(fields: "upc") {
+            upc: String @external
+            rating: Int
+          }
+        `,
+      },
+    ]);
+
+    expect(errors).toBeUndefined();
+    expect(queryPlan).toMatchInlineSnapshot(`
+      QueryPlan {
+        Sequence {
+          Fetch(service: "products") {
+            {
+              topProducts {
+                __typename
+                ... on Car {
+                  name
+                  __typename
+                  upc
+                }
+                ... on Shoe {
+                  name
+                }
+                ... on Shoe @include(if: true) {
+                  __typename
+                  upc
+                }
+              }
+            }
+          },
+          Flatten(path: "topProducts.@") {
+            Fetch(service: "reviews") {
+              {
+                ... on Shoe {
```

Comment:
I think strictly taken we'd also want the include condition here. That doesn't matter for this query, but if you select an extension field from one subgraph conditionally and from another unconditionally, you always fetch the key fields but only want to send a representation for the entity if the condition is met. 

I just wanted to mention it because I thought of it, but it's definitely out of scope for this PR. Right now, the code that extracts the field values for a representation ignores directives anyway, so it wouldn't even make a difference if the query plan added them.

Reply:
I did notice that we could have directives there too, though as that part is unrelated to the regression this PR is concerned with, I haven't looked further.

That said, I quickly concluded that this should never matter (at least for include/skip; for other directives, that may) so I'm not sure what case you're thinking of for "if you select an extension field from one subgraph conditionally and from another unconditionally" and I'd be curious if you had an example (and if we do have a case where things are broken because of this, even if we don't fix it right away, pushing at least a reproduction case might be nice).

Reply:
What I was thinking of is a case where for example you ask for `reviews` conditionally on `Book`, and `inStock` unconditionally. That means you'd be requesting `isbn` unconditionally from the product service. But you may still want to filter books out when collecting representations to send to the reviews subgraph (and avoid a subquery altogether when all top 5 products are books).

```
query {
  topProducts(first: 5) {
    price
    inStock
    ... on Book @skip(if: true) {
      reviews {
        body
      }
    }
    ... on Furniture @skip(if: true) {
      reviews {
        body
      }
    }
  }
}
```